agent/engine/docker_image_manager.go

```diff
@@ -275,7 +276,12 @@ func (imageManager *dockerImageManager) performPeriodicImageCleanup(ctx context.
 }
 
 func (imageManager *dockerImageManager) removeUnusedImages() {
-	seelog.Infof("Begin building map of eligible unused images for deletion")
+	seelog.Debug("Attempting to obtain ImagePullDeleteLock for removing images")
+	ImagePullDeleteLock.Lock()
+	seelog.Debug("Obtained ImagePullDeleteLock for removing images")
+	defer seelog.Debug("Released ImagePullDeleteLock after removing images")
```

Comment:
Maybe group these statements in an anonymous function so its easier to see the order of execution?
```
defer func() { ... }
```